salt/kubernetes-master/init.sls

```diff
@@ -13,11 +13,17 @@ include:
   {% do ip_addresses.append("IP: " + extra_ip) %}
 {% endfor %}
 
+# add some extra names the API server could have
 {% set extra_names = ["DNS: " + grains['fqdn']] %}
 {% for extra_name in pillar['api_server']['extra_names'] %}
   {% do extra_names.append("DNS: " + extra_name) %}
 {% endfor %}
 
+# add some standard extra names from the DNS domain
+{% if salt['pillar.get']('dns:domain', '')|length > 0 %}
```

Comment:
I think this test can be simplified as `{% if salt['pillar.get']('dns:domain') %}`